package.json

```diff
@@ -502,6 +504,27 @@
                     "default": null,
                     "description": "%makefile-tools.configuration.makefile.compileCommandsPath.description%",
                     "scope": "resource"
+                },
+                "makefile.panel.visibility": {
+                    "type": "array",
```

Comment:
Hellp @bobbrow , thanks for the quick replay. My goal is to have a list of features that need to be toggled independently of each other. Being a list, I thought of using an array. The goal being able to add optional (toggle-able features) [easily](https://github.com/jdmchp/vscode-makefile-tools/blob/main/src/configuration.ts#L103). The original PR had a single optional feature (build/debug locally).

Reply:
An object would be a better representation of what you want.  If you do an array of objects, then this would be valid syntax in the settings:
```
{
   "makefile.panel.visibility": [
    {
      "enableLocalDebugging": false
    },
    {
      "enableLocalDebugging": true
    }
  ]
}
```
It would be better to just do an object:
```
{
   "makefile.panel.visibility": {
     "enableLocalDebugging": false
   }
}
```

Reply:
The use of an array made [iterating](https://github.com/jdmchp/vscode-makefile-tools/blob/main/src/configuration.ts#L138) through the possible different 'option flags' easy. I can refactor to make the `makefile.panel.visibility` and object and then iterate through its properties instead.
Thanks for the input.

Reply:
I updated the PR. Using an object makes the code more readable. So, thanks for that suggestion.
![pr_291_single_object_controls_multiple_features](https://user-images.githubusercontent.com/100438651/161319925-bbe4b729-5205-48e3-9883-29e8e75f2df7.gif)
I am new to typescript/json. Things flow differently when using java/xml (what I am used to).